lib/ansible/modules/database/aerospike/aerospike_migrations.py

```diff
@@ -0,0 +1,535 @@
+        supports_check_mode=True
+    )
+    if not LIB_FOUND:
+        module.fail_json(
+            msg="A required module was not found. This playbook" +
+            " requires the 'aerospike' 'time' and 're' modules. " +
+            "Please run 'pip install aerospike'. The other modules" +
+            " should be included in a basic python install."
+        )
+
+    if module.check_mode:
+        return result
+
+    migrations = Migrations(module)
+    # migrations.has_migs accepts 1 argument,
+    # a bool to specify checking local node only or entire cluster
+    has_migrations, skip_reason = migrations.has_migs(module.params['local_only'])
+    if not has_migrations:
+        result['message'] = "No migrations"
+    else:
+        result['message'] = skip_reason
+        module.fail_json(
+            msg="Failed.", **result
+        )
+
+    module.exit_json(**result)
+
+
+class Migrations:
+    """ Check or wait for migrations between nodes """
+
+    def __init__(self, module):
+        self.module = module
+        self._client = self._create_client().connect()
+        self._nodes = {}
+        self._update_nodes_list()
+        self._cluster_statistics = {}
+        self._update_cluster_statistics()
+        self._namespaces = set()
+        self._update_cluster_namespace_list()
+        self._build_list = set()
+        self._update_build_list()
+        self._start_cluster_key = \
+            self._cluster_statistics[self._nodes[0]]['cluster_key']
+
+    # TODO: add support for auth, tls, and other special features
+    # I won't use those features, so I'll wait until somebody complains
+    # or does it for me (Cross fingers)
+    # create the client object
+    def _create_client(self):
+        config = {
+            'hosts': [
+                (self.module.params['host'], self.module.params['port'])
+            ],
+            'policies': {
+                'timeout': self.module.params['connect_timeout']
+            }
+        }
+        return aerospike.client(config)
+
+    # delimiter is for seperate stats that come back, NOT for kv
+    # separation which is =
+    def _info_cmd_helper(self, cmd, node=None, delimiter=';'):
+        if node is None:  # If no node passed, use the first one (local)
+            node = self._nodes[0]
+        data = self._client.info_node(cmd, node)
+        data = data.split("\t")
+        if len(data) != 1 and len(data) != 2:
+            self.module.fail_json(
+                msg="Unexpected number of values returned in info command: " +
+                str(len(data))
+            )
+        # data will be in format 'command\touput'
+        data = data[1]
+        data = data.rstrip("\n\r")
+        data_arr = data.split(delimiter)
+
+        # some commands don't return in kv format
+        # so we dont want a dict from those.
+        if '=' in data:
+            retval = dict(
+                metric.split("=") for metric in data.split(delimiter))
```

Comment:
What happens if you have something like `a=b=c`? Then you will init the dictionary with the tuple `(a, b, c)`, which will fail.

Usually you want to interpret `a=b=c` as key `a` with value `b=c`; for that, you need `metric.split("=", maxsplit=1)`.

Reply:
Also, why do you compute `data.split(delimiter)` again instead of using `data_arr`?

```suggestion
                metric.split("=", maxsplit=1) for metric in data_arr)
```

Reply:
This is actually a scenario for some of the already exposed metrics. They aren't used by this module, but very good feedback. I will add it! Thanks for the tip, didn't know about this.

Reply:
Probably a brain fart on the data_arr lol